format/spec.md

```diff
@@ -841,19 +855,45 @@ Notes:
 
 ## Delete Formats
 
-This section details how to encode row-level deletes in Iceberg delete files. Row-level deletes are not supported in v1.
+This section details how to encode row-level deletes in Iceberg delete files. Row-level deletes are added by v2 and are not supported in v1. Deletion vectors are added in v3 and are not supported in v2 or earlier. Position delete files must not be added to v3 tables, but existing position delete files are valid.
```

Comment:
Shall we add some notes that the first ever operation after migrating table to V3 must produce a DV that includes all previous position deletes for that file? And that implementations can safely discard all file-scoped delete files but have to keep partition-scoped delete files.

Reply:
I don't think that this is the right place for it. The spec should state requirements and these points are covered in more specific requirements. This states the requirement to not add non-DV position delete files to v3 tables, and the section on delete vectors states the requirement that when you write a DV, it must include all existing position deletes for the file:

> If a DV is written for a data file, it must replace all previously written position delete files so that when a DV is present, readers can safely ignore matching position delete files.

There's a good opportunity to add context down in the v3 section that outlines the changes though. I'll add a note there.

Reply:
There was already quite a bit in that section, but I added the point about partition-scoped delete files:

> Position delete files that contain deletes for more than one data file need to be kept in table metadata until all deletes are replaced by DVs